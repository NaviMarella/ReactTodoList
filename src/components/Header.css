.header{
    background-color: #E6AC27;
    color: white;
    min-height: 60px;
    display: flex;
    flex-direction: row;
    align-items: center;
    width: 100vw;
}

.header_title_container{
    display: flex;
    flex: 2;
    justify-content: center;
     align-items: center;
}

.header_logo, .hamLogo{
    height: 50px;
    width: 50px;
    margin: 0 20px;
}

.hamLogo{
    color: #339194;
}