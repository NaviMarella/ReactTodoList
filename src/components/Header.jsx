import PropTypes from 'prop-types';
import React from "react";
import hamburgerIcon from "../icons/hamburger.svg";
import todoIcon from "../icons/todo.svg";
import "./Header.css";

const Header = ({showOrHideSidebar}) => {

    return (
        <header className="header">
            <img alt="" onClick={() => showOrHideSidebar()} className={["header_logo", "hamLogo"].join(" ")} src={hamburgerIcon}></img>
            <div className="header_title_container">
                <img alt="" className="header_logo" src={todoIcon}></img>
                <h1>Todoist</h1>
            </div>
        </header>
    )
}

Header.propTypes = {
    showOrHideSidebar: PropTypes.func.isRequired
};

export default Header;