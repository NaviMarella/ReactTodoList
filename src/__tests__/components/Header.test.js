import { shallow } from 'enzyme';
import React from 'react';
import Header from './../../components/Header';

describe('Header Component', () => {
  test('render snapshot', () => {
      let props = {
        showOrHideSidebar: jest.fn()
      }
    const wrapper = shallow(<Header {...props} />);
     wrapper.find('img').at(0).simulate('click');
     expect(props.showOrHideSidebar).toHaveBeenCalled();

    expect(wrapper).toMatchSnapshot();
  });
});
