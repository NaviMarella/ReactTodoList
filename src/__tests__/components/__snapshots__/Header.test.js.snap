// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Component render snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Header
    showOrHideSidebar={
      [MockFunction] {
        "calls": Array [
          Array [],
        ],
        "results": Array [
          Object {
            "isThrow": false,
            "value": undefined,
          },
        ],
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <img
          alt=""
          className="header_logo hamLogo"
          onClick={[Function]}
          src="hamburger.svg"
        />,
        <div
          className="header_title_container"
        >
          <img
            alt=""
            className="header_logo"
            src="todo.svg"
          />
          <h1>
            Todoist
          </h1>
        </div>,
      ],
      "className": "header",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "alt": "",
          "className": "header_logo hamLogo",
          "onClick": [Function],
          "src": "hamburger.svg",
        },
        "ref": null,
        "rendered": null,
        "type": "img",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <img
              alt=""
              className="header_logo"
              src="todo.svg"
            />,
            <h1>
              Todoist
            </h1>,
          ],
          "className": "header_title_container",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "alt": "",
              "className": "header_logo",
              "src": "todo.svg",
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Todoist",
            },
            "ref": null,
            "rendered": "Todoist",
            "type": "h1",
          },
        ],
        "type": "div",
      },
    ],
    "type": "header",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <img
            alt=""
            className="header_logo hamLogo"
            onClick={[Function]}
            src="hamburger.svg"
          />,
          <div
            className="header_title_container"
          >
            <img
              alt=""
              className="header_logo"
              src="todo.svg"
            />
            <h1>
              Todoist
            </h1>
          </div>,
        ],
        "className": "header",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "",
            "className": "header_logo hamLogo",
            "onClick": [Function],
            "src": "hamburger.svg",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <img
                alt=""
                className="header_logo"
                src="todo.svg"
              />,
              <h1>
                Todoist
              </h1>,
            ],
            "className": "header_title_container",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "alt": "",
                "className": "header_logo",
                "src": "todo.svg",
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Todoist",
              },
              "ref": null,
              "rendered": "Todoist",
              "type": "h1",
            },
          ],
          "type": "div",
        },
      ],
      "type": "header",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
